### DISCUSSION TITLE
Disable popup close button or make popup triggers stateful.

### DISCUSSION THREAD

**Discussion by pgale** [Clooos/Bubble-Card#600] [Feature requests]
I use a series of popups on my Lenovo ThinkSmart View panels. They work pretty well. I do find that the popup close button:

1) Can break the logic of the Pop-up Trigger - if using a boolean to trigger the popup, pressing the close button or navigating to another popup will result in the popup becoming out of sync with the boolean. Ideally, the popup should alter the state of the boolean if it's closed by any of the other methods allowed - stateful I think it's called.

2) Gets in the way and uses valuable screen real estate if not needed. I'm happy using the horizontal stack buttons to open/close the popups or via the bottons within my popups.

Is it possible through CSS etc to remove the close button completely as a work-around?

**Comment by zyi840449** [comment]
type: vertical-stack
view_layout:
  position: sidebar
cards:
  - type: custom:bubble-card
    card_type: pop-up
    hash: '#bath_sensor'
    button_type: name
    margin_top_mobile: 220px
    bg_opacity: '30'
    shadow_opacity: '50'
    bg_blur: '5'
    **styles: |
      .pop-up > :first-child
      {display: none !important}**


Do what I do, and you will find that the X button is gone

**Reply by pgale** [reply]
excellent many thanks. I'll give that a go and it might fix part of my issue.

I'd still like to see proper stateful popus linked to a boolean though :)

**Reply by pgale** [reply]
Got it:
``` 
styles: |
  .pop-up > :first-child
  {display: none !important}
```

**Reply by pgale** [reply]
works well thanks :)

**Comment by pgale** [comment]
Moved to issues.
